src/nccl_ofi_rdma.c

```diff
@@ -1936,10 +1953,13 @@ static inline int free_base_req(uint64_t *num_inflight_reqs,
 		goto exit;
 	}
 
+	elem = req->elem;
+
 	/* Zero out buffer */
 	zero_nccl_ofi_req(req);
 
-	nccl_ofi_freelist_entry_free(nccl_ofi_reqs_fl, req);
+	assert(elem);
```

Comment:
why?

Reply:
Within this function, we don't know that `elem` is not NULL.

We do assert the arguments to `nccl_ofi_freelist_entry_free` are not NULL, so I suppose we can remove this assert.

Reply:
Removed